assignment-3/part-2/server.js

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env node
+
+// create a simple cli tool to benchmark a server using commander library
+// following options can be used:
+// - `-c[concurrency]` - number of parallel requests to perform at a time,
+// - `-n[requests]` - number of requests to perform for the benchmarking session,
+// - `-b[body]` - if specified, should sent a random generated body with request,
+// - `url` - url which should be used for requests.
+
+const http = require('http');
+const program = require('commander');
+const axios = require('axios');
+const chalk = require('chalk');
+
+program
+  .version('0.0.1')
+  .option(
+    '-c, --concurrency <n>',
+    'number of parallel requests to perform at a time',
+    1
+  )
+  .option(
+    '-n, --requests <n>',
+    'number of requests to perform for the benchmarking session',
+    2
+  )
+  .option(
+    '-b, --body [type]',
+    'if specified, should sent a random generated body with request'
+  )
+  .option(
+    'url',
+    'url which should be used for requests',
+    'https://www.google.com/'
+  );
+
+program.parse(process.argv);
+
+const { concurrency, requests: req, body, url } = program.opts();
+const postData = JSON.stringify({
+  msg: 'Hello World!',
+});
+
+const timeArr = [];
+
+async function makeRequests(c) {
+  for (let i = 0; i < req; i = i + c) {
```

Comment:
it does a single request, since `c` is not parsed from string. `i` becomes NaN on a second iteration

Reply:
Ohh yes I was trying to add the parallel requests but it did not work I forgot to remove the `c` argument from the function.

Reply:
@korzio I think its fixed now atleast the Nan part. I get the following output when I try.
![image](https://user-images.githubusercontent.com/43214045/177736747-74e34020-e963-42e7-9e6b-2d1f5c786b77.png)